client/src/App.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import { HashRouter, Route, Redirect, Switch } from 'react-router-dom';
+import $ from 'jquery';
+import './global-jquery'; // bootstrap-material-design needs global jQuery
+import 'bootstrap-material-design/dist/js/material';
+import 'bootstrap-sass/assets/javascripts/bootstrap/modal';
+import Flash from './Flash';
+import LoginForm from './LoginForm';
+import NoteEdit from './NoteEdit';
+import './App.css';
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isLoggedIn: undefined
+    }
+  }
+
+  render() {
+    return (
+      <HashRouter>
+        <div className="container-fluid">
+          <Flash />
+          { this.isLoggedIn() === true ?
+            <Switch>
+              <Route path='/notes/:id' component={ NoteEdit } />
+              <Route path='/notes' component={ NoteEdit } />
+              <Redirect from='/' to='/notes' />
+            </Switch>
+            : null
+          }
+          { this.isLoggedIn() === false ?
+            <div className="row">
+              <div className="col-md-6 col-md-offset-3">
+                <div className="well">
+                  <LoginForm onLoginSuccess={this.onLoginSuccess.bind(this)} />
+                </div>
+              </div>
+            </div>
+            : null
+          }
```

Comment:
you could also do something like:

```
  { this.isLoggedIn() === true ? this.renderLoggedIn() : this.renderLoggedOut() }

```

and then 

```

renderLoggedIn()  {
  return (
 <Switch>
    <Route path='/notes/:id' component={ NoteEdit } />
    <Route path='/notes' component={ NoteEdit } />
     <Redirect from='/' to='/notes' />
   </Switch>
) }
```

Reply:
or defining components for that

Reply:
Updated (in a similar way to your suggestion).